### DISCUSSION TITLE
Rasterio multispectral graphs showing up blank?

### DISCUSSION THREAD

**Discussion by kholl03** [giswqs/geog-312#86] [Q&A] [answered]
Hi all. I'm currently trying to graph stacked bands, however, they're showing up blank. I get an error when I open the file. Could it be that I'm opening it wrong? Please see attached screenshots.  

<img width="1440" alt="Screenshot 2024-10-02 at 3 58 43 PM" src="https://github.com/user-attachments/assets/4acac436-1261-43ee-b34a-66b79a8c516f">
<img width="1440" alt="Screenshot 2024-10-02 at 3 58 53 PM" src="https://github.com/user-attachments/assets/590e14d3-ed4d-4658-8ff9-8ee287e8f3d3">

**Comment by giswqs** [comment, marked as answer]
Follow the example at https://geog-312.gishub.org/book/geospatial/rasterio.html#stacking-and-plotting-bands.

**Reply by Iamnyatichi** [reply]
I also have the same problem. I have followed the example in the lecture notes but I still get the same output as shown below.
![image](https://github.com/user-attachments/assets/8c887c83-8545-40ee-86ec-4d5ab180b5ab)

**Comment by decorouz** [comment]
If you are still having challenges, I would suggest you normalize the bands(array). 
```py
def normalize(array):
    arr_min, arr_max = array.min(), array.max()
    return (array - arr_min)/(arr_max - arr_min)
```

**Reply by kholl03** [reply]
Hey! This is what I ended up trying and it worked like a charm. Thank you!

**Reply by Stelios-Ervis-Priftis** [reply]
Hey guys, I was having the same issue as well and this helped a lot, thank you!

So after the normalization, the expected image result should look like the image below, right?

<img width="704" alt="Screenshot 2025-03-25 at 21 04 03" src="https://github.com/user-attachments/assets/56f82a2e-d04d-42d2-9ae4-c2fabca0e612" />
